### ISSUE TITLE
Ball got stuck top left corner

### ISSUE THREAD

**Issue by seabastion** [dozingcat/Vector-Pinball#7]
I didn't manage to get a screenshot but somehow the ball got stuck on the "default" table up at the top left corner just to the right of the white circle.

**Comment by notfoss** [comment]
I just encountered the same! It is a very edge case probably. The ball just stayed there and I had to quit the game and start again to fix the problem.

And I managed to get a screenshot :grinning: 

<a href='http://postimg.org/image/fhnw5w4xf/' target='_blank'><img src='http://s29.postimg.org/fhnw5w4xf/Screenshot_2015_03_01_15_45_56.jpg' border='0' alt="Screenshot 2015 03 01 15 45 56" /></a>

**Comment by notfoss** [comment]
The ball got stuck again, but this time it got stuck on the divider next to the right save.

<a href='http://postimg.org/image/cpmt30man/' target='_blank'><img src='http://s11.postimg.org/cpmt30man/Screenshot_2015_12_04_23_03_11.jpg' border='0' alt="Screenshot 2015 12 04 23 03 11" /></a>

**Comment by SamSkulls** [comment]
It'd be cool if we could use the accelerometer to tilt the machine, eh?

**Comment by notfoss** [comment]
I was thinking of the exact same thing! In fact, the first time this happened, I instinctively tried tilting the phone to see if something happens :p

**Comment by SamSkulls** [comment]
Found this in field.java. Looks like most of the necessary code has already been written. :o

```java
    /** Adjusts gravity in response to the device being tilted; not currently used. */
    public void receivedOrientationValues(float azimuth, float pitch, float roll) {
        double angle = roll - Math.PI/2;
        float gravity = layout.getGravity();
        float gx = (float)(gravity * Math.cos(angle));
        float gy = -Math.abs((float)(gravity * Math.sin(angle)));
        world.setGravity(new Vector2(gx, gy));
    }
```

**Comment by dozingcat** [comment]
Right, I haven't enabled tilting because avoiding false positives can be tricky. But feel free to play around with the code and let me know if you come up with something good.

I would like to at least detect the case of the ball getting stuck and then "auto-bump" it.

**Comment by dozingcat** [comment]
Commit 9f35df62b503f4dbf863d658f0ffcbf8dcb4e664 attempts to detect when the ball is stuck and nudge it.

**Comment by notfoss** [comment]
That's great! But considering that the ball getting stuck happens so rarely, we might have to wait for a long time before we can confirm that the fix works. But I'll make sure to post a comment here.

On a side note, can you bump the version, so that I can get it from f-droid?

**Comment by jaxankey** [comment]
If it doesn't otherwise affect the game play, I'd recommend bumping to Google play too. This has happened to me many times, especially on the many-color-small-bumpers board where there are two places the ball can rest on a bumper if it's slow enough. Believe it or not, I actually got it stuck on the single-pixel-wide line between the side alleys. Does this mean I play too much?

Anyway, here's hoping the bump is is a random direction. A vertical bump would return it to its original position in the last case.

**Comment by jaxankey** [comment]
(Note also the bump should be big enough to set off the bumpers in case it is wedged)

**Comment by SecUpwN** [comment]
Hi there, everyone! First of all, huge thanks @dozingcat for developing this neat app and especially for publishing it on F-Droid! About this Issue: Ha, the same thing happened to me the other day, I did take a screenshot and just wanted to file this when I noticed that @seabastion already did. If you ask me, I'd vote for implementing the tilt as suggested in https://github.com/dozingcat/Vector-Pinball/issues/7#issuecomment-162807752, of course including setting off the bumpers. This even opens up the possibility to add a funky a noise alarm when shaking the phone.

**Comment by jaxankey** [comment]
Yeah, the "shake the phone" option could work, just requiring a very high
threshold / violent shake to bump the ball and set off the bumpers.

On Wed, Jan 13, 2016 at 12:50 AM, Security: Pwned. <notifications@github.com

**Comment by dozingcat** [comment]
Version 1.5.2 with the auto-bump fix should be available the Play store now.

**Comment by SecUpwN** [comment]
>Version 1.5.2 with the auto-bump fix should be available the Play store now.

I`m a privacy-geek and only update via F-Droid. Hope to see it there soon.
